.github/workflows/test-linux.yml

```diff
@@ -0,0 +1,125 @@
+---
+name: Test Linux
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: >
+          Python version to currently test
+        type: string
+        required: true
+      runner:
+        description: >
+          Describes the system this workflow should run on.
+        type: string
+        required: true
+      use-msrv:
+        description: >
+          Decides whether we use MSRV or stable for rust toolchain.
+        required: false
+        type: boolean
+        default: false
+      install-optionals:
+        description: >
+          Decides whether we install optional dependencies
+        type: boolean
+        default: false
+      install-from-sdist:
+        description: >
+          Decides whether we perform rust tests
+        type: boolean
+        default: false
+
+jobs:
+  unit-tests:
+    if: github.repository_owner == 'Qiskit'
+    name: ${{ inputs.runner }}-tests-Python-${{ inputs.python-version }}
+    runs-on: ${{ inputs.runner }}
+    timeout-minutes: 60
+    env:
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@1.79
+        if: ${{ inputs.use-msrv }}
+      - name: Set up Python ${{ inputs.python-version }}
+        id: python-outputs
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+      - name: Cache stestr
+        uses: actions/cache@v4
+        with:
+          key: 'stestr | "${{ runner.os }}" | "${{ inputs.python-version }}" | "${{ github.run_id }}"'
+          restore-keys: |
+            stestr | "${{ runner.os }}" | "${{ inputs.python-version }}"
+            stestr | "${{ runner.os }}"
+            stestr
+          path: .stestr
+      - name: Prepare venv
+        run: |
+          set -e
+          python -m pip install --upgrade pip setuptools wheel
+          python -m venv test-job
+      - name: Install Qiskit from sdist
+        if: ${{ inputs.install-from-sdist }}
+        run: |
+          set -e
+          # Use stable Rust, rather than MSRV, to spot-check that stable builds properly.
+          rustup override set stable
```

Comment:
If we were doing this (which we shouldn't in this PR) wouldn't this need to be conditioned on `use-msrv` not install from sdist?

Reply:
Removed in 8d54c24